sorc/fvcom_tools.fd/module_ncio.f90

```diff
@@ -1,2553 +1,2545 @@
-                                  replace_var_nc_real_1d,replace_var_nc_real_2d,      &
-                                  replace_var_nc_real_3d,                             &
-                                  replace_var_nc_int_1d,replace_var_nc_int_2d,        &
-                                  replace_var_nc_int_3d,                              &
-                                  replace_var_nc_char_1d,replace_var_nc_char_2d,      &
-                                  replace_var_nc_char_3d
-      procedure :: replace_var_nc_int
-      procedure :: replace_var_nc_int_1d
-      procedure :: replace_var_nc_int_2d
-      procedure :: replace_var_nc_int_3d
-      procedure :: replace_var_nc_real
-      procedure :: replace_var_nc_real_1d
-      procedure :: replace_var_nc_real_2d
-      procedure :: replace_var_nc_real_3d
-      procedure :: replace_var_nc_double
-      procedure :: replace_var_nc_double_1d
-      procedure :: replace_var_nc_double_2d
-      procedure :: replace_var_nc_double_3d
-      procedure :: replace_var_nc_char
-      procedure :: replace_var_nc_char_1d
-      procedure :: replace_var_nc_char_2d
-      procedure :: replace_var_nc_char_3d
-
-      procedure :: handle_err
-
-      procedure :: convert_theta2t_2dgrid
-!Add a new 3d variable to output file (David.M.Wright)
-      procedure :: add_new_var => add_new_var_3d 
+     character(len=256) :: filename !< Name of data file.
+     integer :: ncid !< File ID.
+     integer :: status !< Return code.
+     integer :: debug_level !< Debug level.
+
+     integer :: nDims !< Number of dims.
+     integer :: ends(4) !< Counts of dims.
+     integer :: xtype !< Type of data.
+     character(len=40) :: dimname(4) !< Name of dims.
+   contains
+     procedure :: open => open_nc !< Open netCDF file. @return
+     procedure :: close => close_nc !< Close netCDF file. @return
+     procedure :: get_dim => get_dim_nc !< read in dimension from the nc file @return
+     generic   :: get_att => get_att_nc_int,get_att_nc_real,get_att_nc_string !< Get attribute. @return
+     procedure :: get_att_nc_int !< Get attribute. @return
+     procedure :: get_att_nc_real !< Get attribute. @return
+     procedure :: get_att_nc_string !< Get attribute. @return
+     generic   :: get_var => get_var_nc_double_1d, get_var_nc_double_2d, & 
+          get_var_nc_double_3d,                       &
+          get_var_nc_real_1d,get_var_nc_real_2d,      &
+          get_var_nc_real_3d,                         &
+          get_var_nc_short_1d,get_var_nc_short_2d,    &
+          get_var_nc_int_1d,get_var_nc_int_2d,        &
+          get_var_nc_int_3d,                          &
+          get_var_nc_char_1d,get_var_nc_char_2d,      &
+          get_var_nc_char_3d      !< Read in a 1d, 2d, 3d, or 4d field from the nc file. @return
+     procedure :: get_var_nc_short !< Read in a 1d, 2d, 3d, or 4d field from the nc file. @return
```

Comment:
The description is the same for all these 'get_var' procedures?  Do they do the same thing?

Reply:
Some of the 'replace_var' routines also have the same description.

Reply:
@GeorgeGayno-NOAA No, each one should be slightly different. For each get_var_nc_YYYY_Xd procedure, it will call the corresponding get_var_nc_YYYY the appropriate number of times to read in the Xd variable of type YYYY. For example, get_var_nc_char_2d will call get_var_nc_char the correct number of times needed to fully read in the 2d char-type field.

Reply:
Ok.  I was confused by the doxygen comments.

Reply:
As mentioned elsewhere, the doxygen bug is going to prevent good documentation here. So these descriptions are just to silence the doxygen warnings.

There is good documentation above each function, which notes the differences between functions.

We will hope for a better handling of type enclosed functions in a future version of doxygen.